unittests/linuxliketests.py

```diff
@@ -1704,8 +1702,14 @@ def test_prelinking(self):
                            stderr=subprocess.DEVNULL,
                            text=True, timeout=1)
         obj_files = p.stdout.strip().split('\n')
-        self.assertEqual(len(obj_files), 1)
-        self.assertTrue(obj_files[0].endswith('-prelink.o'))
+        if is_osx():
+            # There are at least two -- one for the prelinked object,
+            # another for the symbol list
+            self.assertGreaterEqual(len(obj_files), 1)
+            self.assertTrue(any([o.endswith('-prelink.o') for o in obj_files]))
```

Comment:
```suggestion
            self.assertTrue(any(o.endswith('-prelink.o') for o in obj_files))
```

using a list comprehension instead of a generator expression means that the whole list must be created before any is called, while the generator expression can exit early.